dispatch/apps/api/views.py

```diff
@@ -143,6 +144,12 @@ def get_queryset(self):
             # If a search term (q) is present, filter queryset by term against `full_name`
             queryset = queryset.filter(full_name__icontains=q)
         return queryset
+    
+    def perform_destroy(self, instance):
+        try:
+            instance.delete()
+        except ProtectedError:
+            raise APIException('Cannot delete PERSON because attached to USER')
```

Comment:
After taking a second look at this, I think it would make more sense to raise a custom exception in this case.

You can create a new exception class by extending `APIException` in `dispatch/apps/api/exceptions.py`. There's an example in that file already.

In this case I think it makes the most sense to return a `HTTP_409_CONFLICT` status code to imply that the person can't be deleted due to a conflict with another resource.